utils/post.js

```diff
@@ -341,8 +344,12 @@ const modifyNewItemAnonymity = (newItem, isBlurredPost = false) => {
   return newItem;
 };
 
-const isUserFollowAuthor = async (newItem, followingUsers) => {
+const isUserFollowAuthor = async (newItem, followingUsers, karmaScores = []) => {
   if (newItem?.actor?.id) {
+    if (karmaScores.length > 0) {
+      const user = karmaScores.find((user) => user.user_id === newItem?.actor?.id);
+      newItem.karma_score = roundingKarmaScore(user?.karma_score || 0);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `isUserFollowAuthor` function has been modified to include karma score calculation. However, there is no error handling in case the `roundingKarmaScore` function fails or returns an unexpected value. It would be better to add some error handling here.
```diff
  if (karmaScores.length > 0) {
    const user = karmaScores.find((user) => user.user_id === newItem?.actor?.id);
+   try {
      newItem.karma_score = roundingKarmaScore(user?.karma_score || 0);
+   } catch (error) {
+     console.error(`Error calculating karma score: ${error}`);
+     newItem.karma_score = 0;
+   }
  }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->